safe_cli/main.py

```diff
@@ -118,11 +122,39 @@ def loop(self):
                 pass
 
 
-def build_safe_cli():
+def get_safe_from_owner(
+    owner: ChecksumAddress, node_url: str
+) -> Optional[ChecksumAddress]:
+    """
+    Show a list of Safe to chose between them and return the selected one.
+    :param owner:
+    :param node_url:
+    :return: Safe address of a selected Safe
+    """
+    ethereum_client = EthereumClient(node_url)
+    safe_tx_service = TransactionServiceApi.from_ethereum_client(ethereum_client)
+    safes = safe_tx_service.get_safes_for_owner(owner)
+    if len(safes):
+        # Show safes
+        for option, safe in enumerate(safes):
+            print_formatted_text(HTML(f"{option} - <b>{safe}</b> "))
+        option = choose_option_question(
+            "Select the Safe to initialize the safe-cli", len(safes)
+        )
+        if option is not None:
+            return safes[option]
+
+    print_formatted_text(
+        HTML(f"<ansired>No safe was found for the specified owner {owner}</ansired>")
+    )
+    return None
```

Comment:
Maybe all this together should be part of `choose_option_question`, or we can create a new method `choose_option_from_list_question`, as it seems that having a list and selecting a number is something common in our code (we are doing it also for ledger)